### ISSUE TITLE
[BUG] No physical devices found

### ISSUE THREAD

**Issue by minecrawler** [amethyst/amethyst#1632]
## Description

When trying to run an example, I get the above panic.

```
cargo run --example material
```

## Screenshots and Logs

```
    Finished dev [unoptimized + debuginfo] target(s) in 9m 39s
     Running `target\debug\examples\material.exe`
[INFO][winit::platform::platform::window] Guessed window DPI factor: 1
[INFO][amethyst::app] Initializing Amethyst...
[INFO][amethyst::app] Version: 0.10.0
[INFO][amethyst::app] Platform: x86_64-pc-windows-gnu
[INFO][amethyst::app] Amethyst git commit: 1e7bd29511ee711c4053f85be82827fb7b7f4920
[INFO][amethyst::app] Rustc version: 1.33.0 Stable
[INFO][amethyst::app] Rustc git commit: 2aa4c46cfdd726e97360c2734835aa3515e8c858
[WARN][rendy_factory::factory] Slow safety checks are enabled! Disable them in production by enabling the 'no-slow-safet
y-checks' feature!
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: ErrorMessage { msg: "No physical devices found"
}', src\libcore\result.rs:997:5
note: Run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
error: process didn't exit successfully: `target\debug\examples\material.exe` (exit code: 101)
```

## Your Environment

1. Operating System Windows
2. Operating System Version 10_64
3. Version of Amethyst master (1e7bd29511ee711c4053f85be82827fb7b7f4920)

## Additional Context

Tested on several two different computers with different hardware. Both, however, use Intel/NVIDIA hardware and Win10_64.

**Comment by jaynus** [comment]
You need to provide the feature flag now to be passed. 

cargo run --example <> --features "vulkan" 
cargo run --example <> --features "metal" 
etc.

When consuming amethyst in a project, you now need something like:
`amethyst = { git = "https://github.com/amethyst/amethyst", features = ["nightly", "vulkan" ]}`

**Comment by vikaig** [comment]
I had the same problem, maybe it should be added to the readme?

**Comment by minecrawler** [comment]
From the thread on the forum:

I am very positive that a lot more people will come up with it, because it is not intuitive to add extra parameters when running something. Also, the error we is very bad, because it does not help finding the problem at all, but looks like something went wrong when searching for a graphics card. Here’s what I propose:

1. Set a sane default rendering path, for example OpenGL
2. If no rendering API is selected, fail early and literally tell the user that they have to select one via feature-gate

**Comment by torkleyy** [comment]
We can re-open this, but it's already being addressed in #1635
